infrastructure/terraform/environments/staging/coredns.tf

```diff
@@ -0,0 +1,91 @@
+locals {
+  coredns_cluster_ip = "10.83.240.20"
+}
+
+resource "helm_release" "coredns" {
+  name  = "coredns"
+  repository = "https://coredns.github.io/helm"
+  chart = "coredns"
+
+  depends_on = [
+    google_container_node_pool.primary_spot_nodes
+  ]
+
+  namespace = "kube-system"
+  create_namespace = false
+
+  set {
+    name = "replicaCount"
+    value = 2
+  }
+
+  set {
+    name = "service.clusterIP"
+    value = local.coredns_cluster_ip
+  }
+}
+
+resource "kubectl_manifest" "coredns_config_map" {
+  depends_on = [
+    helm_release.coredns
+  ]
+
+  yaml_body = <<YAML
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: coredns-coredns
```

Comment:
from what I can see the only reason to deploy this is the different name (`coredns` vs `coredns-coredns`). Can we just use a variable for this in the cluster resources and set it differently for staging, or will this cause issues with GKE? Having the same config at two places will undoubtedly lead to inconsistencies.

Reply:
This was a dependency issue. You cannot create the ConfigMap for CoreDNS before the completion of the release. Fixed!